vllm/model_executor/models/molmo.py

```diff
@@ -1210,7 +1223,16 @@ def get_input_embeddings(
     ) -> torch.Tensor:
         inputs_embeds = self.model.get_input_embeddings(input_ids)
         if multimodal_embeddings is not None:
-            inputs_embeds = inputs_embeds + multimodal_embeddings
+            assert isinstance(multimodal_embeddings, (list, tuple))
+            multimodal_embeddings = torch.cat(multimodal_embeddings, dim=0)
+            diff = inputs_embeds.shape[0] - multimodal_embeddings.shape[0]
+            if diff:
+                # We need to pad at the front of the multimodal embeddings to
+                # take input ids from other running requests into account.
+                inputs_embeds = inputs_embeds + F.pad(multimodal_embeddings,
+                                                      (0, 0, diff, 0))
```

Comment:
The padding logic here assumes that the decoding requests will be at the front of the batch. If this is not always the case please let me know! @WoosukKwon

Reply:
As we discussed in the slack, this is not always the case 😢

Reply:
Hmm yea, then I wonder how to proceed from here - we probably need to take a look and see what information we can get from model runner to make this work.

Reply:
Actually - I will check their input processor code on HF to see if I can find anything useful

Reply:
@WoosukKwon This has been resolved by [0176b7b](https://github.com/vllm-project/vllm/pull/10699/commits/0176b7b4618423cbe120a85db91ef4c9b4d28255)